flask/app/buckets.py

```diff
@@ -17,54 +17,80 @@
 @login_required
 def get_unlinked_files():
 
-    minio_client = get_minio_client()
-
-    app.logger.debug("Getting all minio bucket names..")
-    all_bucket_names = [bucket.name for bucket in minio_client.list_buckets()]
-
-    # Remove buckets the current user does not have access to assuming group_code.lower() == bucket name
-    app.logger.debug("Getting all buckets that user has access to..")
-    user = (
-        models.User.query.filter_by(user_id=current_user.user_id)
-        .options(joinedload(models.User.groups))
-        .first()
-    )
-
-    valid_bucket_names = []
-    for group in user.groups:
-        code = group.group_code.lower()
-        if code in all_bucket_names:
-            valid_bucket_names.append(code)
-
-    app.logger.debug("Getting all files in valid minio buckets..")
-    all_files = []
-    for bucket in valid_bucket_names:
-        objs = minio_client.list_objects(bucket, recursive=True)
-        for obj in objs:
-            all_files.append(bucket + "/" + obj.object_name)
-
-    app.logger.debug("Getting all linked files..")
+    path = request.args.get("path", type=str)
 
     files = []
 
-    linked_files = {
-        f.path: True
-        for f in models.File.query.filter(
-            or_(models.File.multiplexed == None, models.File.multiplexed == False)
-        ).all()
-    }
-
-    linkable_files = {
-        f.path: True
-        for f in models.File.query.filter(models.File.multiplexed == True).all()
-    }
-
-    for file_name in all_files:
-        if linkable_files.get(file_name):
-            files.append({"path": file_name, "multiplexed": True})
-        elif not linked_files.get(file_name):
-            files.append({"path": file_name, "multiplexed": False})
-
-    app.logger.debug("Returning JSON array..")
+    if path != "None":
+        filepath = path.split("/", 1)
+        if len(filepath) < 2:
+            abort(
+                400,
+                description="Please make sure your file path contains both folder and file name, such as c4r/folder1/folder2/filename",
```

Comment:
Please call the buckets and prefixes for what they are, otherwise it could lead to confusion.

Reply:
This comment wasn't addressed!

Reply:
To clarify, I changed the variable names "bucket" to "bucket_name" and "prefix" to "file_name". Was this not what you meant?

Reply:
I was looking at the error message here.

Reply:
Ok, cool. Just changed the error message!